mail/src/main/java/javax/mail/internet/MimeMessage.java

```diff
@@ -2207,6 +2207,7 @@ public void saveChanges() throws MessagingException {
      * @since		JavaMail 1.4
      */
     protected void updateMessageID() throws MessagingException {
+    if(getHeader("Message-ID") == null)
 	setHeader("Message-ID", 
```

Comment:
I think it would be better practice to put braces around the setHeader statement. Also the code change is short enough it could all fit on one line.

Reply:
Single line is too long to fit all the code.  So, using the multiple lines.